From da28fd70f97e1439f11ae37f32e1ef13756a1fcb Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 6 Oct 2020 14:35:10 -0700
Subject: [PATCH 06/16] UBUNTU: SAUCE: apparmor: drop prefixing abs root labels
 with '='

This prefix was not ever meant to go to the logs, and is just a debug
hint. Unfortunately it is showing up in the logs resulting in a double
'=' eg.

  subj==firefox

which has the potential to break some log parsing tools.

Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Andrea Righi <andrea.righi@canonical.com>
Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 security/apparmor/label.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index e68bcedca976..464766cf1ef2 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -1632,13 +1632,10 @@ int aa_label_snxprint(char *str, size_t size, struct aa_ns *ns,
 	AA_BUG(!str && size != 0);
 	AA_BUG(!label);
 
-	if (flags & FLAG_ABS_ROOT) {
+	if (flags & FLAG_ABS_ROOT)
 		ns = root_ns;
-		len = snprintf(str, size, "=");
-		update_for_len(total, len, size, str);
-	} else if (!ns) {
+	else if (!ns)
 		ns = labels_ns(label);
-	}
 
 	label_for_each(i, label, profile) {
 		if (aa_ns_visible(ns, profile->ns, flags & FLAG_VIEW_SUBNS)) {
-- 
2.32.0

